noxfiles/docker_nox.py

```diff
@@ -9,35 +15,36 @@ def get_current_image() -> str:
 
 
 @nox.session()
-def build(session: nox.Session) -> None:
+@nox.parametrize(
+    "image",
+    [
+        nox.param("prod", id="prod"),
+        nox.param("dev", id="dev"),
+        nox.param("test", id="test"),
+        nox.param("ui", id="ui"),
+    ],
+)
+def build(session: nox.Session, image: str) -> None:
     """Build the Docker container for fidesctl."""
+    build_matrix = {
+        "prod": {"tag": get_current_image(), "target": "prod"},
+        "dev": {"tag": IMAGE_LOCAL, "target": "dev"},
+        "test": {"tag": IMAGE_LOCAL, "target": "prod"},
+        "ui": {"tag": IMAGE_LOCAL_UI, "target": "frontend"},
```

Comment:
hmm so I think we might not actually want this, or maybe not in `local-ui`. It's not the worst, but it'll do an extra `npm run export` call. As applications get bigger, this can be slow, and is not actually needed for local environments (since that's the static build, which, if you're doing FE development, you wouldn't need).

it might be the easiest way to get to the node docker image though... the alternative would be to add an extra stage called frontend-build or something like that, which uses `frontend` as the base, then calls `npm run export`, and then this target would still be `frontend`.

Reply:
However, I do think there is a case for having the built static file visible _somewhere_, as the only way to really see it (and not just whatever you have locally) is to comment out the docker-compose mount, as you and I have talked about before. And it looks like fidesops is running into that as well and it is causing confusion https://github.com/ethyca/fidesops/pull/625#issuecomment-1155165054

Reply:
hmmm ok, im still digging into this and will come up with a decent solution

Reply:
@allisonking can you clarify a bit what you're asking for here?

The gist is that I am needing to run an image with node that contains the UI files and this seemed like a logical way to me, but it sounds like you believe there's a better way?

Reply:
so after thinking about it some more, I think it's fine to go with this for now. We would only want to revisit if the next builds start to take a longer time (the last project I worked on, JS builds could take ~5 minutes, so we avoided building it when in local/dev environments. but it was also much larger than what we currently have). 

One way to avoid it (there are probably others, but haven't thought too much about it), would be to add another stage in the Dockerfile

```
####################
## Prepare frontend ##
####################
FROM node:16 as frontend-base

WORKDIR /fides/clients/admin-ui

# install node modules
COPY clients/admin-ui/ .
RUN npm install

####################
## Build frontend ##
####################
FROM frontend-base as frontend-build

# Build the frontend static files
RUN npm run export

...

COPY --from=frontend-build /fides/clients/admin-ui/out/ /fides/src/fidesapi/build/static/
```
And then you would target `frontend-base` instead in nox.

Reply:
@allisonking I'm going to go ahead and merge this for now since I don't want it to get stale + it accomplishes the original task, but I'm going to pop open an issue to add some kind of `prod` command that spins up the database, webserver and UI statically (to mirror a real deployment). I can also pop open an issue to update the dockerfile with an additional stage (I agree that will be necessary eventually)